services/authentication/current_user.rb

```diff
@@ -0,0 +1,15 @@
+class AuthenticationService
+  class CurrentUser < ApplicationService
+    def initialize(env)
+      @env = env
+    end
+
+    def call
+      env['warden'].user
```

Comment:
Memoization would've looked nice here

Reply:
```ruby

    def call
      user
    end

    private
    
    attr_reader :user
    
    def user
      @user ||= env['warden'].user
    end
```
like this? Looks a bit weird imo

Reply:
No, inside of a call method

Reply:
damn, I don't see it. How exactly should it look like?

Reply:
```ruby
def call
  @user ||= env['warden'].user
end
```

Reply:
Hm, okay, but what's the purpose?

Reply:
Increasing the performance during the sequential calls to the same instance

Reply:
@kushniryb 
Does that makes sense inside the service?
He must to save a result of the service somewhere instead of instantiating user inside service, IMO. New service, new user

Reply:
@ailecksandr 
> New service, new user
Yeah, you're right and that's how it'll work. I'm just saying that memoization would be nice in case you were to invoke `call` method a few times in a row on the same service instance.
Does that make sense or am I missing something here?

Reply:
@kushniryb hm. What if user logged out during session. Looks like this example with memoization will keep returning previous user, won't it?

Reply:
In such case there'll be a new instance once the request is complete which will return `nil` or whatever
You should test it though to make sure nothing like that happens